### PR TITLE
Qt: Replace "OK" with "Close" buttons

### PR DESCRIPTION
**Pull request by spycrab**
This PR replaces the "OK" buttons in dialogues with "Close" buttons.

### CODE REVIEW

**Comment by spycrab** [comment]
@dolphin-emu-bot rebuild

**Comment by spycrab** [comment]
@MayImilae

**Comment by BhaaLseN** [comment]
I'd consider this a usability feature; so I'm more leaning towards "No" to this change. It is debatable for the Controller configuration (since it might close the window by accident); but for all others this (and also the Cancel/Close button, most commonly bound to Escape) should stay.

**Comment by spycrab** [comment]
@BhaaLseN 

> I'd consider this a usability feature

Pressing Enter to closed from dialogues that are opened from the main window will result in an unwanted side effect though: Starting the currently selected game. So it isn't very useful at all.

> (and also the Cancel/Close button, most commonly bound to Escape) should stay.

That is unaffected by this PR and still works.

The original reason for this PR is that it was requested by MayImilae (not necessarily the "No Enter" thing but that the "OK" button is not highlighted by default). 

Additionally Wx behaved in the same way (Which isn't a good argument for anything, so feel free to ignore this).

Hope this clears this up a bit.

**Comment by BhaaLseN** [comment]
Hm, maybe? I agree that hitting Enter inside a dialog shouldn't start a game. But this sounds more like a focus issues (ie. the GameList registering it as command) rather than a dialog issue.
I'll leave it up to @MayImilae since she also has a pretty good understanding of UX etc.; don't let my comment block this if it otherwise makes sense for other reasons.

**Comment by MayImilae** [comment]
Huh, that's weird. The OK buttons switched sides? Was that intentional? 

screen shot 2018-03-22 at 3 30 50 am

Anyway, I think this is a good change. Not automatically selecting the OK button helps clumsy people avoid accidentally closing the window, as well as accidentally launching a game. And it gets rid of the weird blue button in macOS, which I'm totally for!

File: Source/Core/DolphinQt2/Config/Mapping/MappingWindow.cpp [hunk 1]
```diff
@@ -109,7 +109,12 @@ void MappingWindow::CreateMainLayout()
   m_main_layout = new QVBoxLayout();
   m_config_layout = new QHBoxLayout();
   m_tab_widget = new QTabWidget();
-  m_button_box = new QDialogButtonBox(QDialogButtonBox::Ok);
+  m_button_box = new QDialogButtonBox;
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/Config/Mapping/MappingWindow.cpp]: Is it not possible to just call button()->setDefault(false) (or setAutoDefault)? It feels like some default Qt logic is being unnecessarily reimplemented.

**Comment by spycrab** [comment]
> Huh, that's weird. The OK buttons switched sides? Was that intentional? I'm used to them being on the right since that's the norm on windows. Of course ok buttons aren't normal on macOS at all, but, eh.

Not really but it'll probably have to stay that way until I find a better solution.

**Comment by leoetlino** [comment]
```
<MayImilae> everything works correct!
<MayImilae> if anything it's a bit better, since before it didn't turn blue when clicked
<MayImilae> now it just behaves as it should
<MayImilae> also it's on the right now, yay!
```

**Pull request merged**